projects/packages/search/src/dashboard/components/record-meter/notice-box.jsx

```diff
@@ -19,10 +25,35 @@ const CLOSE_TO_LIMIT_PERCENT = 0.8; //TODO: currently 'close' is defined as 80%.
  */
 export function NoticeBox( props ) {
 	const notices = [];
+	const [ showNotice, setShowNotice ] = useState( true );
+
+	// deal with localStorage for ensuring dismissed notice boxs are not re-displayed
+	const dismissedNoticesString = localStorage.getItem( 'dismissedNoticeBoxes' )
+		? localStorage.getItem( 'dismissedNoticeBoxes' )
+		: '';
+
+	const dismissedNoticesArray =
+		dismissedNoticesString.length > 0 ? dismissedNoticesString.split( ',' ) : [];
+
+	const dismissNoticeBox = () => {
+		setShowNotice( false );
+
+		if ( dismissedNoticesString ) {
+			if ( dismissedNoticesArray.includes( notices[ 0 ].id ) ) {
+				localStorage.setItem(
+					'dismissedNoticeBoxes',
+					dismissedNoticesString + ',' + notices[ 0 ].id
+				);
+			}
+		} else {
+			localStorage.setItem( 'dismissedNoticeBoxes', notices[ 0 ].id );
```

Comment:
By doing this, the message status of dismissed is remembered permanently, however, what if in the future we need to display the message again? For example, a user is about to exceed record limit this month and he's upgraded to the next tier and then is about to exceed again?

A solution could be, we set a time limit for a notice to show ( in json object and persisted in localStorage ), say a month. If a notice is dismissed, we don't try to show it in a month, and after a month, if proper,  pop them up again.

Reply:
a possbile JSON format could be:

```JSON
{
"1": timestamp,
"4": timestamp
}
```

Reply:
I agree with Jasper on implementing an expiry for notice dismissals. However, I'm also okay with implementing this in a follow-up PR instead of this one 🙂

Reply:
@jsnmoon I was thinking the data structure would be different if we implement this. But since we are not going to release soon, followup PR sounds good to me too.

Reply:
I really like this idea, it makes a lot of sense. Adding it to the list for a followup PR 👍